tests/python_package_test/test_basic.py

```diff
@@ -328,3 +328,42 @@ def check_asserts(data):
         lgb_data.set_weight(sequence)
         lgb_data.set_init_score(sequence)
         check_asserts(lgb_data)
+
+    def test_trees_to_dataframe(self):
+
+        def _imptcs_to_numpy(X, impcts_dict):
+            cols = ['Column_' + str(i) for i in range(X.shape[1])]
+            imptcs = []
+            for col in cols:
+                try:
+                    imptcs.append(impcts_dict[col])
+                except KeyError:
+                    imptcs.append(0.)
+            return np.array(imptcs)
+
+        X, y = load_breast_cancer(True)
+        data = lgb.Dataset(X, label=y)
+        num_trees = 10
+        bst = lgb.train({"objective": "binary"}, data, num_trees)
+        tree_df = bst.trees_to_dataframe()
+        split_dict = (tree_df[~tree_df.split_gain.isnull()]
+                      .groupby('split_feature')
+                      .size()
+                      .to_dict())
+
+        gains_dict = (tree_df
+                      .groupby('split_feature')['split_gain']
+                      .sum()
+                      .to_dict())
+
+        tree_split = _imptcs_to_numpy(X, split_dict)
+        tree_gains = _imptcs_to_numpy(X, gains_dict)
+        mod_split = bst.feature_importance('split')
+        mod_gains = bst.feature_importance('gain')
+        num_trees_from_df = tree_df.tree_index.nunique()
```

Comment:
could you change this to `tree_df['tree_index']` and replace all other `df.{column_name}` type references in a similar way?

This is safer and more future-proof, since `pandas` will not prevent you from creating a column whose name conflicts with a public attribute of `pd.DataFrame` but will prefer the attribute to the column when there is a conflict.

example:

```python
import pandas as pd
df = pd.DataFrame({'index': [1,2,3]})
df.index

# >>> RangeIndex(start=0, stop=3, step=1)
```

Reply:
Sure. I just made this change and pushed.

Reply:
Looks great, thank you! I'll resolve this thread

Reply:
@jameslamb Please update your review status as well. Your requested changes block merging:
> Merging is blocked
Merging can be performed automatically once the requested changes are addressed.